src/client.rs

```diff
@@ -101,21 +103,39 @@ impl Client {
     }
 
     /// Run post request towards API
-    pub(crate) async fn post_request_with_body<B>(
+    pub(crate) async fn post_request_with_body<B: Body>(
         &self,
         endpoint: String,
         body: B,
     ) -> APIResult<Response>
     where
         B: Sized + serde::Serialize,
     {
+        let inner_body = body.to_inner(self.options.api_key.clone());
         self.client
-            .get(self.full_url(endpoint))
+            .post(self.full_url(endpoint))
             .header(CONTENT_TYPE, "application/json")
-            .header(AUTHORIZATION, format!("Bearer {}", self.options.api_key))
-            .body(serde_json::to_string(&body).expect("unwrap here is safe"))
+            .body(inner_body.expect("unwrap here is safe"))
             .send()
             .await
             .map_err(|e| Error::Connection(e.to_string()))
     }
 }
+
+pub(crate) trait Body {
+    fn to_inner(self, api_key: String) -> serde_json::Result<String>;
+}
```

Comment:
If the goal is to have the the `inner` structs wit a `api_key` field could you do something where serde is utilized like this? That way all the structs like `Decide`, `Event` etc. doesn't have to know about having an inner version of itself.
```
struct Body<T> {
  api_key: String,
  #[serde(flatten)]
  inner: T,
}
```

See [this](https://serde.rs/attr-flatten.html) example for more details.

Reply:
probably you'll also need some sort of `T: Serialize` ?

Reply:
Good idea, thanks!